public/pages/AnomalyCharts/utils/__tests__/anomalyChartUtils.test.ts

```diff
@@ -75,3 +81,81 @@ describe('anomalyChartUtils', () => {
     });
   });
 });
+
+describe('anomalyChartUtils function tests', () => {
+  const alertResponse = {
+    response: {
+      alerts: [
+        {
+          id: 'eQURa3gBKo1jAh6qUo49',
+          version: 300,
+          monitor_id: 'awUMa3gBKo1jAh6qu47E',
+          schema_version: 2,
+          monitor_version: 2,
+          monitor_name: 'Example_monitor_name',
+          monitor_user: {
+            name: 'admin',
+            backend_roles: ['admin'],
+            roles: ['all_access', 'own_index'],
+            custom_attribute_names: [],
+            user_requested_tenant: null,
+          },
+          trigger_id: 'bQUQa3gBKo1jAh6qnY6G',
+          trigger_name: 'Example_trigger_name',
+          state: 'ACTIVE',
+          error_message: null,
+          alert_history: [
+            {
+              timestamp: 1617314504873,
+              message: 'Example error emssage',
+            },
+            {
+              timestamp: 1617312543925,
+              message: 'Example error message',
+            },
+          ],
+          severity: 1,
+          action_execution_results: [
+            {
+              action_id: 'bgUQa3gBKo1jAh6qnY6G',
+              last_execution_time: 1617317979908,
+              throttled_count: 0,
+            },
+          ],
+          start_time: 1616704000492,
+          last_notification_time: 1617317979908,
+          end_time: null,
+          acknowledged_time: null,
+        },
+      ],
+      totalAlerts: 1,
+    },
+  };
+  //TODO: change acknowledgedTime and endTime type to allow null values potentially
+  const alertConverted = [
+    {
+      monitorName: 'Example_monitor_name',
+      triggerName: 'Example_trigger_name',
+      severity: 1,
+      state: 'ACTIVE',
+      error: null,
+      startTime: 1616704000492,
+      endTime: null,
+      acknowledgedTime: null,
+    },
+  ] as MonitorAlert[];
+  test('convertAlerts', () => {
+    expect(convertAlerts(alertResponse)).toStrictEqual(alertConverted);
+  });
+  test('generateAlertAnnotations', () => {
+    const alertsConverted = generateAlertAnnotations(alertConverted);
+    expect(alertsConverted).toStrictEqual([
+      {
+        dataValue: 1616704000492,
+        details:
+          'There is a severity 1 alert with state ACTIVE from 03/25/21 08:26:40 PM.',
```

Comment:
I'm a little surprised this passed in the github runner. From my experience most tests related to timestamp -> time zone conversion leads to discrepancies from running locally (e.g., PDT timezone) to github runner env (UTC).

Reply:
Yeah I was struggling with that for a little but I was able to solve this by mocking the Date.now() function so it returns the exact time no matter timezones/actual time `Date.now = jest.fn().mockReturnValue(new Date('2021-06-06T12:33:37.000Z'));` https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/pull/252/files#diff-e37ccf9132358762559158e6dc32cdcbc952aaf4af2150d9bc4c2e129ab1f1d0R59.

Reply:
Ah nice, good solution

Reply:
This actually solved the issue with checking for different .now() times but I guess the github runner ended up running on PST since it didn't fail here but failed with adding the `TZ="Asia/Singapore"` flag when running `test:jest`. I was able to fix timezone issue by adding UTC to be default timezone for all jest tests through the config file and removed one test we were currently skipping cause of it. Other skipped tests I wont address in this PR

Reply:
Great, even better! Hopefully we can close out the rest of those skipped ones soon. I don't think there's many of them.